q2_types/feature_data/_format.py

```diff
@@ -142,142 +142,168 @@ def _validate_(self, level):
     'TSVTaxonomyDirectoryFormat', 'taxonomy.tsv', TSVTaxonomyFormat)
 
 
-class DNAFASTAFormat(model.TextFileFormat):
+class FASTAFormat(model.TextFileFormat):
```

Comment:
This all looks pretty reasonable to me - I haven't done a _deep_ dive, but it looks like this is mostly just shuffling some code into a new base class - is that impression fair? My one question/comment, is that right now this base format, `FASTAFormat` is not what is known as an "importable format" - no one can import data in this format because it isn't associated with a particular semantic type _and_ there are no transformers capable of transforming into a compatible format for any of the existing semantic type/importable format combos out there. I don't see any problem with this, but I just want to confirm that that is in fact the intent. So put another way, `FASTAFormat` is a convenience base class that is intended for extension. Did I miss the mark?

Reply:
Thanks @thermokarst, that is how I meant it, yes. So whenever one wants to introduce a new FASTA-based format, one can just extend the base one while explicitly passing a new alphabet.

About your impressions: yes, it's mostly moving the code + introducing the new `AlignedFASTAFormatMixin` that can turn any FASTA format into its _aligned_ version.